File: tracker.py
import tweepy

# Enter your API credentials here
consumer_key = ''
consumer_secret = ''

access_token = ''
access_token_secret = ''

# Set up the Tweepy API object
auth = tweepy.OAuthHandler(consumer_key, consumer_secret)
auth.set_access_token(access_token, access_token_secret)
api = tweepy.API(auth)

# Define a function to handle incoming tweets
class TweetCounter(tweepy.Stream):
    def __init__(self, keyword):
        self.count = 0
        super().__init__(consumer_key, consumer_secret, access_token, access_token_secret)

        self.keyword = keyword

    def on_status(self, status):
        if self.keyword.lower() in status.text.lower():
            self.count += 1
            print("Number of tweets with {}: {}".format(self.keyword, self.count))


# Start the stream
keyword = input("Enter a keyword or hashtag to track: ")
tweet_counter = TweetCounter(keyword)
tweet_counter.filter(track=[keyword])

File: tracker.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <curl/curl.h>

#define CONSUMER_KEY "QUEYiuMpCelese0971Z6rztWd"
#define CONSUMER_SECRET "rXemLt1ngFiHP4RLsjnJgAbcqQ10wWXUl0J8Y07eQ3NPuWDXWM"
#define ACCESS_TOKEN "1372164021160382465-2CjURk6ch80n7Qok2YaaJGbTjJcCZi"
#define ACCESS_TOKEN_SECRET "0JMHDt3uLt6fKGQGAt37BhYxvtHHrE4H6lvrkKSw9GHjm"

void process_tweet(const char* tweet) {
    // Process the received tweet
    printf("Tweet: %s\n", tweet);

    // Perform your analysis here
    // Count the number of tweets containing the keyword
    // Measure the reach of the tweet (e.g., retweets, likes, replies)
}

size_t write_callback(void* contents, size_t size, size_t nmemb, void* userp) {
    size_t real_size = size * nmemb;
    char* tweet = (char*)malloc(real_size + 1);
    memcpy(tweet, contents, real_size);
    tweet[real_size] = '\0';

    process_tweet(tweet);

    free(tweet);
    return real_size;
}

int main() {
    CURL* curl = curl_easy_init();
    if (curl) {
        // Set up Twitter API credentials
        curl_easy_setopt(curl, CURLOPT_USERNAME, CONSUMER_KEY);
        curl_easy_setopt(curl, CURLOPT_PASSWORD, CONSUMER_SECRET);
        curl_easy_setopt(curl, CURLOPT_HTTPAUTH, CURLAUTH_BASIC);

        // Set up Twitter API endpoint for streaming
        const char* url = "https://stream.twitter.com/1.1/statuses/filter.json";
        curl_easy_setopt(curl, CURLOPT_URL, url);

        // Set up request parameters for keyword tracking
        const char* keyword = "kenya";
        const char* post_fields = "track=kenya";
        curl_easy_setopt(curl, CURLOPT_POSTFIELDS, post_fields);

        // Set up write callback function
        curl_easy_setopt(curl, CURLOPT_WRITEFUNCTION, write_callback);

        // Start streaming tweets
        CURLcode res = curl_easy_perform(curl);
        if (res != CURLE_OK) {
            fprintf(stderr, "curl_easy_perform() failed: %s\n", curl_easy_strerror(res));
        }

        // Clean up
        curl_easy_cleanup(curl);
    }

    return 0;
}
